lib/JSONAPISerializer.js

```diff
@@ -13,6 +13,13 @@ const {
 
 const { validateOptions, validateDynamicTypeOptions, validateError } = require('./validator');
 
+// Cache of strings to convert to their converted values per conversion type
+const convertCaseMap = {
```

Comment:
Should the `convertCaseMap` variable be initialized when calling serialize/deserialize instead of being global?

can this have an impact on memory consumption over time when serializing too much different data?

Reply:
Interesting question, I did it this way on purpose to increase performance. I would not expect this to have a large impact of memory, as it is just a collection of strings.

The way I think about it, if a user requests a resource one, it is very likely that the same resource will be requested again. Thus needed to convert the string again if convertCase is specified. Doing it this way ensures that you never compute a conversion for a string more than once during the lifetime of the service.

Reply:
I thought the performance improvement will be applied to a large array of the same resource at serialization/deserialization, not during the lifetime of the service.

Reply:
Maybe the solution here is to use a more complex caching mechanism like LRU cache (Least Recent Usage) or it can free the memory after a timeout.

Reply:
It could be done either way, this way was more performant. If a user makes a lot of small requests, making it not global would not have an impact on the performance, since it needs to recompute everything every time.

Would you agree more with putting the map into the scope of the class?

Having the cache per call to serialize/deserialize I think would not have as much of an impact, except for large payloads.

Reply:
Using an LRU cache would also work well, are you cool with this package?
https://www.npmjs.com/package/lru-cache

Reply:
Maybe we should have a default max number of strings in the cache, and allow the user to override the default on initialization of the class?

Reply:
In an other PR, i try to make this library browser compatible... It will not work anymore with this library. 

Maybe an other possible solution, is to have `transform` option already requested on #70 #66  that will act as a hook before (de)serializing attributes. So you will be able "precompile" the case of your choice:


```javascript
transform(data) {
    return {
      id: data.id
      first-name: data.firstName,
      last-name: data.lastName
      // ...
    }
  }
```

I don't know, if it will be faster or not.

Reply:
I would prefer to not do something like that. Wouldn't that require that you know what all of the attributes are going to be before serialization? There is another package that I was using before this one that did something like that, where you were required to give the attributes that you want serialized ahead of time. This was not very performant and took a lot to compute what was needed.

Reply:
Perhaps we can use this to avoid including a dependency: http://jsfiddle.net/2baax9nk/5/